core/tauri/src/scope/fs.rs

```diff
@@ -79,11 +79,33 @@ fn push_pattern<P: AsRef<Path>, F: Fn(&str) -> Result<Pattern, glob::PatternErro
 ) -> crate::Result<()> {
   let path: PathBuf = pattern.as_ref().components().collect();
   list.insert(f(&path.to_string_lossy())?);
-  #[cfg(windows)]
+  #[cfg(any(windows, target_os = "android"))]
   {
-    if let Ok(p) = std::fs::canonicalize(&path) {
+    let mut path = path;
+    let mut buf = None;
+
+    // attempt to canonicalize parents in case we have a path like `/data/user/0/appid/**`
+    // where `**` obviously does not exist but we need to canonicalize the parent
+    let canonicalized = loop {
+      if let Ok(p) = path.canonicalize() {
+        break Some(if let Some(buf) = buf { p.join(buf) } else { p });
+      }
+
+      let last = path.iter().rev().next().map(PathBuf::from);
+      if let Some(mut p) = last {
+        path.pop();
+        if let Some(buf) = &buf {
+          p.push(buf);
+        }
+        buf.replace(p);
+      } else {
+        break None;
+      }
+    };
```

Comment:
I'm having a hard time following this code logic tbh so might help adding more comments explaining a bit

Reply:
pushed